mod/item.php

```diff
@@ -992,16 +1005,35 @@ function item_post(&$a) {
 		// Insert an item entry for UID=0 for global entries
 		// We have to remove or change some data before that,
 		// so that the post appear like a regular received post.
-		unset($datarray['self']);
-		unset($datarray['wall']);
-		unset($datarray['origin']);
-
-		if (in_array($datarray['type'], array("net-comment", "wall-comment")))
-			$datarray['type'] = 'remote-comment';
-		elseif ($datarray['type'] == 'wall')
-			$datarray['type'] = 'remote';
-
-		add_shadow_entry($datarray);
+		// Additionally there is some data that isn't a database field.
+		$arr = $datarray;
+
+		$arr['app'] = $arr['source'];
+		unset($arr['source']);
+
+		unset($arr['self']);
+		unset($arr['wall']);
+		unset($arr['origin']);
+		unset($arr['api_source']);
+		unset($arr['message_id']);
+		unset($arr['profile_uid']);
+		unset($arr['post_id']);
+		unset($arr['dropitems']);
+		unset($arr['commenter']);
+		unset($arr['return']);
+		unset($arr['preview']);
+		unset($arr['post_id_random']);
+		unset($arr['emailcc']);
+		unset($arr['pubmail_enable']);
+		unset($arr['category']);
+		unset($arr['jsreload']);
+
+		if (in_array($arr['type'], array("net-comment", "wall-comment")))
```

Comment:
While the above quote remark and the spacing around operators aren't part of PSR-2, this is definitely part of it.